src/main/kotlin/ecommerce/controller/CartController.kt

```diff
@@ -1,62 +1,44 @@
 package ecommerce.controller
 
 import ecommerce.dto.auth.AuthenticatedUser
-import ecommerce.dto.cart.AddToCartRequest
-import ecommerce.dto.cart.UpdateQuantityRequest
 import ecommerce.model.Cart
+import ecommerce.model.CartItem
+import ecommerce.service.CartItemService
 import ecommerce.service.CartService
-import jakarta.validation.Valid
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
-import org.springframework.web.bind.annotation.PostMapping
-import org.springframework.web.bind.annotation.PutMapping
-import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RestController
-import java.net.URI
 
-@RequestMapping("/api")
+@RequestMapping("/api/carts")
 @RestController
 class CartController(
     private val cartService: CartService,
+    private val cartItemService: CartItemService,
 ) {
-    @GetMapping("/cart-items")
-    fun getCartItems(user: AuthenticatedUser): List<Cart> {
-        return cartService.getCartItems(user.userId)
-    }
-
-    @PostMapping("/cart-items")
-    fun addToCart(
-        @Valid @RequestBody addToCartRequest: AddToCartRequest,
-        user: AuthenticatedUser,
-    ): ResponseEntity<Cart> {
-        val cart = cartService.addToCart(user.userId, addToCartRequest)
-        return ResponseEntity.created(URI.create("/api/cart-items")).body(cart)
+    @GetMapping("/{userId}")
+    fun getCart(
+        @PathVariable userId: Long,
+    ): Cart {
+        return cartService.getCartByUserId(userId)
     }
 
-    @PutMapping("/cart-items/{productId}")
-    fun updateQuantity(
-        @PathVariable productId: Long,
-        @Valid @RequestBody updateRequest: UpdateQuantityRequest,
+    @GetMapping("/{cartId}/items")
+    fun getAllCartItemsOfCart(
+        @PathVariable cartId: Long,
```

Comment:
It seems like it would be more appropriate to place this in `CartItemController`! 🙂